### PR TITLE
SRVCLI-401: :gift: Add kn-event configuration

### PR DESCRIPTION
**Pull request by cardil**
## Changes

 - :gift: Add kn-event configuration

Fixes [SRVCLI-401](https://issues.redhat.com/browse/SRVCLI-401)

Supersedes openshift-knative/hack#154
Requires openshift-knative/kn-plugin-event#540
Requires openshift-knative/kn-plugin-event#542
Requires openshift-knative/kn-plugin-event#543

### CODE REVIEW

File: config/kn-plugin-event.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
         enabled: true
       openShiftVersions:
         - version: '4.16'
           useClusterPool: true
+    release-1.16:
+      konflux:
+        enabled: true
+      openShiftVersions:
+        - version: '4.17'
+          useClusterPool: true
```
Comment (mgencur) [inline config/kn-plugin-event.yaml:14]: AFAIK, we don't have a cluster pool for 4.17 yet. The version here should be 4.16 for now. When we bump the cluster pool we can increase the version in this place and in other repository configs as well.

Comment (cardil) [reply config/kn-plugin-event.yaml:14]: I backed down it to 4.16 cluster pool in 02e43dad1961425861c9406246f1d19f2d4ee911 for now.

File: .github/workflows/release-generate-ci.yaml [hunk 2]
```diff
@@ -430,12 +437,12 @@ jobs:
...
-        name: '[kn-plugin-func - release-v1.15] Create Konflux PR'
+        name: '[kn-plugin-event - release-1.15] Create Konflux PR'
         run: |
           set -x
-          repo="kn-plugin-func"
-          branch="sync-konflux-release-v1.15"
-          target_branch="release-v1.15"
+          repo="kn-plugin-event"
+          branch="sync-konflux-release-1.15"
+          target_branch="release-1.15"
```
Comment (cardil) [inline .github/workflows/release-generate-ci.yaml:445]: This looks weird. The func config shouldn't be replaced by the event one :confused:.

File: Makefile [hunk 3]
```diff
@@ -35,6 +35,10 @@ generate-functions-ci:
 	go run github.com/openshift-knative/hack/cmd/prowgen --config config/kn-plugin-func.yaml $(ARGS)
 .PHONY: generate-functions-ci
 
+generate-event-ci:
```
Comment (mgencur) [inline Makefile:38]: Can you rename this to `generate-kn-plugin-event-ci` ? We already have `generate-eventing-ci`. It's confusing

File: .github/workflows/release-generate-ci.yaml (hunk 2 above)
Comment (cardil) [reply .github/workflows/release-generate-ci.yaml]: It looks like it was fixed in 2b5bdc51dcd3c954290f1f88772fa4ae33d4c6a4

Something must have been confused while resolving the conflict.

File: Makefile (hunk 3 above)
Comment (cardil) [reply Makefile:38]: The func plugin got `generate-functions-ci`, so I just followed that.

Maybe, we can change the function's target as well to have some consistency: `make generate-plugin-func-ci`

The `make generate-kn-plugin-event-ci` is so mouthful!

BTW. this convention is stuttering - `make generate-`. Better would be `make event-plugin`, `make eventing` etc.

Comment (mgencur) [reply Makefile:38]: At minimum let's just rename your new target to generate-plugin-event-ci

**Pull request merged**